Core/NoOpRunner.Core/GameWindow.cs

```diff
@@ -66,18 +66,32 @@ public void ClickShape(int x, int y)
                 }
             }
 
-            for (int i = 0; i <= windowPixels.GetUpperBound(0); i++)
+            // save space and time that give speed or breakup
+            // for (int i = 0; i <= windowPixels.GetUpperBound(0); i++) 
+            // {
+            //     for (int j = 0; j <= windowPixels.GetUpperBound(1); j++)
+            //     {
+            //         if (windowPixels[i, j] == default)
+            //         {
+            //             windowPixels[i, j] = new WindowPixel(i, j, Enums.Color.Black, false);
+            //         }
+            //     }
+            // }
+
+            return windowPixels;
+        }
+
+        public IEnumerable<WindowPixel> GetCurrentWindowEnumerable()//generator no?
```

Comment:
Well, a multidim array is more convenient to access. Generator would make sense if there were cases where you would only need access to some of the pixels of the current window (i.e. the first 10), or maybe if you wanted to process all the pixels sequentially (convenient LINQ queries). Can you think of any such situations? And even if there are such cases, it'd make more sense to implement the generation version only when we actually come to them - otherwise YAGN applies. Your current use just gets the full list anyways, so it doesn't make too much sense to impl this.

Reply:
Window pixels save row and column numbers inside itself, could use enum, 'cuz UIElementsContainer is list and removing out of bounds pixels by x and y easier and faster than redrawing all pixels. Using enum and sparse matrix is equal in this situation .